Sources/Foundation/Process.swift

```diff
@@ -1134,14 +1134,17 @@ open class Process: NSObject {
     open func waitUntilExit() {
         let runInterval = 0.05
         let currentRunLoop = RunLoop.current
-        let checkRunLoop : () -> Bool = (currentRunLoop == self.runLoop)
-                ? { currentRunLoop.run(mode: .default, before: Date(timeIntervalSinceNow: runInterval)) }
-                : { currentRunLoop.run(until: Date(timeIntervalSinceNow: runInterval)); return true }
 
-        // update .runLoop to allow early wakeup.
+        let runRunLoop : () -> Void = (currentRunLoop == self.runLoop)
+                ? { currentRunLoop.run(mode: .default, before: Date(timeIntervalSinceNow: runInterval)) }
```

Comment:
The current run loop may be configured to return upon a source process, and it could be returning `kCFRunLoopRunHandledSource`, terminating the loop before the process completed.  We would now simply pump the runloop while we believe the child is executing, which I believe was the original intention.